frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -347,17 +351,47 @@ end
 
 function ReaderTypeset:onSetPageHorizMargins(h_margins)
     self.unscaled_margins = { h_margins[1], self.unscaled_margins[2], h_margins[2], self.unscaled_margins[4] }
-    self:onSetPageMargins(self.unscaled_margins)
+    self.ui:handleEvent(Event:new("SetPageMargins", self.unscaled_margins))
 end
 
 function ReaderTypeset:onSetPageTopMargin(t_margin)
     self.unscaled_margins = { self.unscaled_margins[1], t_margin, self.unscaled_margins[3], self.unscaled_margins[4] }
-    self:onSetPageMargins(self.unscaled_margins)
+    if self.sync_t_b_page_margins then
+        self.unscaled_margins[4] = t_margin
+        -- Let ConfigDialog know so it can update it on screen and have it saved on quit
+        self.ui.document.configurable.b_page_margin = t_margin
+    end
+    self.ui:handleEvent(Event:new("SetPageMargins", self.unscaled_margins))
 end
 
 function ReaderTypeset:onSetPageBottomMargin(b_margin)
     self.unscaled_margins = { self.unscaled_margins[1], self.unscaled_margins[2], self.unscaled_margins[3], b_margin }
-    self:onSetPageMargins(self.unscaled_margins)
+    if self.sync_t_b_page_margins then
+        self.unscaled_margins[2] = b_margin
+        -- Let ConfigDialog know so it can update it on screen and have it saved on quit
+        self.ui.document.configurable.t_page_margin = b_margin
+    end
+    self.ui:handleEvent(Event:new("SetPageMargins", self.unscaled_margins))
+end
+
+function ReaderTypeset:onSyncPageTopBottomMargins(toggle)
```

Comment:
You went the extra mile there ;).

I probably wouldn't have made it somewhat "retroactive" like that, instead just relying on what the user would next touch (top or bottom) and sync to that, but I get the idea, and hey, why the hell not ;).

Plus, it should essentially be a NOP with default values anyway ;).

Reply:
Yes :) And it's quite useful to rebalance the margin tweaks.

It ended up being an unexpected bonus in my workflow:
- start with them sync'ed, see what you can get
- if you only get the missing line when they are recuded too much (I like some top margin), do one step back (so, missing line back), and increase just the top margin till things get centered)